### ISSUE TITLE
wrong behavior for 'default' argument in parser.add_boolean()

### ISSUE THREAD

**Issue by natechols** [PacificBiosciences/pbcommand#11]
In pbcommand/models/parser.py, the "default" argument for add_boolean() has opposite effects on the tool contract parser and argparser.  In the TC parser, "default=False" means that the value will be False if the parameter is not set in the resolved tool contract.  In the argparser, default=False is equivalent to action="store_false", making the actual default value True if the argument is unused.

**Comment by pb-cdunn** [comment]
I don't understand. If someone forgets `action='store_true'`, then it's not a boolean. But otherwise, behavior seems to be as expected. Could you show an example?

``` python
import argparse

def parse(parser, args):
    a = parser.parse_args(args)
    print('args=%r => %r' %(args, a))

parser = argparse.ArgumentParser()
parser.add_argument('-b',
        action='store_true',
        default=False)
parse(parser, [])
parse(parser, ['-b'])

parser = argparse.ArgumentParser()
parser.add_argument('-b',
        action='store_false',
        default=True)
parse(parser, [])
parse(parser, ['-b'])

parser = argparse.ArgumentParser()
parser.add_argument('-b',
        default=False)
parse(parser, [])
parse(parser, ['-b something'])
```

``` sh
$ python foo.py -b
args=[] => Namespace(b=False)
args=['-b'] => Namespace(b=True)
args=[] => Namespace(b=True)
args=['-b'] => Namespace(b=False)
args=[] => Namespace(b=False)
args=['-b something'] => Namespace(b=' something')
```

**Comment by natechols** [comment]
I'm describing what happens in the wrapper around argparse that pbcommand provides - see implementation here, where "default=True" is translated into "action='store_true'" (but not otherwise passed to argparse):

https://github.com/PacificBiosciences/pbcommand/blob/master/pbcommand/models/parser.py#L313

**Comment by pb-cdunn** [comment]
:ok_hand:

**Comment by mpkocher** [comment]
Nat, does this fix work for you?

``` python
def add_boolean(self, option_id, option_str, default, name, description):
        d = {True: "store_true", False: "store_false"}
        opt = '--' + option_str
        self.parser.add_argument(opt, action=d[_validate_argparse_bool(not default)],
                                 help=description)
```

This is not particularly obvious how it works from the TC, but it's the right behavior.

If no provided commandline option is provided the parsed default will be the default value. Adding --my-option will enable the opposite of the default value. The option in RTC and TC have the expected behavior, so they don't need to be changed.

**Comment by natechols** [comment]
Yes, this was exactly what I had in mind.

**Comment by mpkocher** [comment]
Fixed in 0.2.1
